include/aikido/control/ros/detail/util-impl.hpp

```diff
@@ -0,0 +1,40 @@
+#include <chrono>
+#include <mutex>
+#include <ros/ros.h>
+#include <ros/callback_queue.h>
+#include <actionlib/client/action_client.h>
+
+namespace aikido {
+namespace control {
+namespace ros {
+
+template <class ActionSpec, class TimeoutDuration, class PeriodDuration>
+bool waitForActionServer(
+  actionlib::ActionClient<ActionSpec>& actionClient,
+  ::ros::CallbackQueue& callbackQueue,
+  TimeoutDuration timeoutDuration = std::chrono::milliseconds{ 1000 },
+  PeriodDuration periodDuration = std::chrono::milliseconds{ 10 }
+)
+{
+  using Clock = std::chrono::steady_clock;
+
+  const auto startTime = Clock::now();
+  const auto endTime = startTime + timeoutDuration;
+  auto currentTime = startTime + periodDuration;
```

Comment:
This should not have `periodDuration` added onto it. Otherwise, the first iteration will have twice the expected period.